### ISSUE TITLE
Does not work on non-Linux, non-Docker platforms

### ISSUE THREAD

**Issue by ltning** [Amonith/mattermost-file-list#18]
The plugin ships with precompiled binaries for specific platforms, and no build instructions for other platforms (FreeBSD). Is there anything about this code that makes it hard or impossible to run on other platforms?

**Comment by Amonith** [comment]
Not really, I don't think so. It's just that the plugin template provided by Mattermost doesn't support FreeBSD and others as they are not supported platforms for Mattermost altogether:

https://docs.mattermost.com/install/requirements.html#mattermost-server-operating-system

You can try editing those lines:

https://github.com/Amonith/mattermost-file-list/blob/master/Makefile#L82

and running the makefile (make dist).
You need to have NPM and GO installed.
You can also find more info here:

https://github.com/mattermost/mattermost-plugin-starter-template
